app/src/main/kotlin/io/imhungry/model/RestaurantEntity.kt

```diff
@@ -0,0 +1,12 @@
+package io.imhungry.model
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Entity(tableName = "restaurant")
+data class RestaurantEntity (
+    @ColumnInfo(name = "name") val name: String
+) {
+    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") var id: Int = 0
```

Comment:
This needs to be in the constructor and not the body of the class.

There should be no default value of it, it will be provided by room.

Reply:
Moving the id field to the constructor requires that an id be passed each time a RestaurantEntity is created. Is this what we want?

Reply:
Looking into it a bit, it seems you are right. You can have the default value of 0, but you should still move it to the constructor. Apparently if you set it to 0 during insertion, it will still auto-generate. It may be worth testing that for confirmation, though.